ethereum/sidecar/server.go

```diff
@@ -264,53 +267,54 @@ func RunServer(
 		server.logger.Info("gRPC server routine stopped")
 	}()
 
-	server.logger.Info(
-		"Waiting for the initial AssetsLocked sync before launching " +
-			"AssetsUnlocked routine",
-	)
+	id, err := server.bridgeContract.ValidatorIDs(accountAddress)
+	if err != nil {
+		panic(fmt.Sprintf("failed to get bridge validator ID: %v", err))
+	}
 
-	// Wait until the initial synchronization of the AssetsLocked routine is
-	// complete.
-	select {
-	case <-server.assetsLockedReady:
+	if id != 0 {
 		server.logger.Info(
-			"Initial AssetsLocked sync completed; proceeding with " +
-				"AssetsUnlocked routines",
+			"Account is an attesting validator. Waiting for the initial " +
+				"AssetsLocked sync before launching AssetsUnlocked routine",
 		)
-	case <-ctx.Done():
-		server.logger.Info(
-			"Context canceled while waiting; exiting without launching " +
-				"AssetsUnlocked routines",
-		)
-		return
-	}
 
-	go func() {
-		// TODO: Only a subset of validators should be attesting AssetsUnlocked
-		//       events. Decide whether we should run this goroutine by calling
-		//       MezoBridge.bridgeValidators() with our Ethereum address.
-
-		defer cancelCtx()
-		err := server.observeAssetsUnlockedEvents(ctx)
-		if err != nil {
-			server.logger.Error(
-				"AssetsUnlocked events observation routine failed",
-				"err", err,
+		// Wait until the initial synchronization of the AssetsLocked routine is
+		// complete.
+		select {
+		case <-server.assetsLockedReady:
+			server.logger.Info(
+				"Initial AssetsLocked sync completed; proceeding with " +
+					"AssetsUnlocked routines",
+			)
+		case <-ctx.Done():
+			server.logger.Info(
+				"Context canceled while waiting; exiting without launching " +
+					"AssetsUnlocked routines",
 			)
+			return
 		}
 
-		server.logger.Info("AssetsUnlocked events observation routine stopped")
-	}()
+		go func() {
+			defer cancelCtx()
+			err := server.observeAssetsUnlockedEvents(ctx)
+			if err != nil {
+				server.logger.Error(
+					"AssetsUnlocked events observation routine failed",
+					"err", err,
+				)
+			}
 
-	go func() {
-		// TODO: Only a subset of validators should be attesting AssetsUnlocked
-		//       events. Decide whether we should run this goroutine by calling
-		//       MezoBridge.bridgeValidators() with our Ethereum address.
+			server.logger.Info("AssetsUnlocked events observation routine stopped")
+		}()
 
-		defer cancelCtx()
-		server.attestAssetsUnlockedEvents(ctx)
-		server.logger.Info("AssetsUnlocked events attestation routine stopped")
-	}()
+		go func() {
+			defer cancelCtx()
+			server.attestAssetsUnlockedEvents(ctx)
+			server.logger.Info("AssetsUnlocked events attestation routine stopped")
+		}()
+	} else {
+		server.logger.Info("Account is NOT an attesting validator")
```

Comment:
```suggestion
		server.logger.Info("sidecar does not represent a bridge validator; skipping AssetsUnlocked events processing")
```

Reply:
Done in b9f834dd9372274d9bd5e7fe4faeb6e7748b6a18